org.envirocar.app/src/org/envirocar/app/views/login/SignupActivity.java

```diff
@@ -351,7 +351,7 @@ private boolean checkUsernameValidity(String username) {
         } else if (username.length() < 6) {
             usernameEditText.setError(getString(R.string.error_invalid_username));
             isValidUsername = false;
-        } else if (!Pattern.matches(USERNAME_REGEX,username)) {
+        } else if (!Pattern.matches(USERNAME_REGEX,username.trim())) {
```

Comment:
Be sure that the email is not only trimmed for checking the pattern, but also for signing up the user. Otherwise, it may happen that the email address will be stored with leading or trailing spaces.